packages/react-dom/src/events/ReactDOMEventListener.js

```diff
@@ -264,62 +362,55 @@ export function dispatchEvent(
   );
 }
 
-// Attempt dispatching an event. Returns a SuspenseInstance or Container if it's blocked.
-export function attemptToDispatchEvent(
+// Returns a SuspenseInstance or Container if it's blocked.
+// Returns null if not blocked and we should use closestInstance
+// Returns nullTarget if not blocked but we should dispatch without a targetInst
+export function findInstanceBlockingEvent(
   domEventName: DOMEventName,
   eventSystemFlags: EventSystemFlags,
   targetContainer: EventTarget,
   nativeEvent: AnyNativeEvent,
-): null | Container | SuspenseInstance {
+): NullTarget | null | Container | SuspenseInstance {
   // TODO: Warn if _enabled is false.
 
   const nativeEventTarget = getEventTarget(nativeEvent);
-  let targetInst = getClosestInstanceFromNode(nativeEventTarget);
+  const targetInst = getClosestInstanceFromNode(nativeEventTarget);
 
   if (targetInst !== null) {
     const nearestMounted = getNearestMountedFiber(targetInst);
-    if (nearestMounted === null) {
-      // This tree has been unmounted already. Dispatch without a target.
-      targetInst = null;
-    } else {
+    if (nearestMounted !== null) {
       const tag = nearestMounted.tag;
       if (tag === SuspenseComponent) {
         const instance = getSuspenseInstanceFromFiber(nearestMounted);
         if (instance !== null) {
           // Queue the event to be replayed later. Abort dispatching since we
           // don't want this event dispatched twice through the event system.
-          // TODO: If this is the first discrete event in the queue. Schedule an increased
-          // priority for this boundary.
           return instance;
         }
         // This shouldn't happen, something went wrong but to avoid blocking
         // the whole system, dispatch the event without a target.
         // TODO: Warn.
-        targetInst = null;
+        return nullTarget;
       } else if (tag === HostRoot) {
         const root: FiberRoot = nearestMounted.stateNode;
         if (root.isDehydrated) {
           // If this happens during a replay something went wrong and it might block
           // the whole system.
           return getContainerFromFiber(nearestMounted);
         }
-        targetInst = null;
+        return nullTarget;
       } else if (nearestMounted !== targetInst) {
         // If we get an event (ex: img onload) before committing that
         // component's mount, ignore it for now (that is, treat it as if it was an
         // event on a non-React tree). We might also consider queueing events and
         // dispatching them after the mount.
-        targetInst = null;
+        return nullTarget;
       }
+    } else {
+      // This tree has been unmounted already. Dispatch without a target.
+      return nullTarget;
```

Comment:
I couldn't come up with a test case where we `findInstanceBlockingEvent` would return `nullTarget`. It doesn't seem possible that it would ever happen because it would mean that there is a React Instance defined on the target but then `nearestMountedFiber` returns `null` for this case. I added a test which attempted to simulate this by removing the target node from its react tree into a tree where there is no react root but that didn't seem to work.  

`getClosestInstanceFromNode(getEventTarget(queuedEvent.nativeEvent))` did return `null` for this case though.

Based off how hydration works it shouldn't be possible for the `nearestMountedFiber` to ever be anything other than a `SuspenseInstance`. Unless the node is manually moved in userland to a different tree thats already hydrated, I tried doing that but that also didn't trigger the `nullTarget`. I'm unsure what else to try to trigger it. Any ideas?